src/ILCompiler.CppCodeGen/src/CppCodeGen/CppWriter.cs

```diff
@@ -1033,6 +1039,11 @@ private String GetCodeForObjectNode(ObjectNode node, NodeFactory factory, bool g
             if (node is ISymbolDefinitionNode)
             {
                 offset = (node as ISymbolDefinitionNode).Offset;
+
+                // Method generic dictionaries get prefixed by the hash code of the owning method
+                if (node is MethodGenericDictionaryNode)
+                    offset -= _compilation.TypeSystemContext.Target.PointerSize;
```

Comment:
`MethodGenericDictionaryNode.HeaderSize` ?

Reply:
Or even better - would an unconditional call to `ISymbolDefinitionNode.Offset` work ?

Reply:
`ISymbolDefinitionNode.Offset` points to the part of data after header but we doesn't know header size. I think using `MethodGenericDictionaryNode.HeaderSize` is more suitable, but it's necessary to make it public.